### PR TITLE
Seq-optimization clean version

### PR DESCRIPTION
**Pull request by KhalilCK**
This PR is a clean up version of https://github.com/goblint/analyzer/pull/965 and contains only the Seq changes.

### CODE REVIEW

File: src/analyses/base.ml [hunk 1]
```diff
@@ -107,8 +107,8 @@ struct
         | (info,value)::xs ->
           match value with
           | `Address t when hasAttribute "goblint_array_domain" info.vattr ->
-            let possibleVars = PreValueDomain.AD.to_var_may t in
-            List.fold_left (fun map arr -> VarMap.add arr (info.vattr) map) (pointedArrayMap xs) @@ List.filter (fun info -> isArrayType info.vtype) possibleVars
+            let possibleVars = List.to_seq (PreValueDomain.AD.to_var_may t )in
```
Comment (michael-schwarz) [inline src/analyses/base.ml:110]: Whitespace is a bit wonky here...

File: src/analyses/base.ml [hunk 2]
```diff
@@ -1257,7 +1257,7 @@ struct
             | _ -> None
           in
           let alen = List.filter_map (fun v -> lenOf v.vtype) (AD.to_var_may a) in
-          let d = List.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) (List.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int) (slen @ alen)) in
+          let d = Seq.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) ((Seq.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int))@@ List.to_seq (slen @ alen)) in
```
Comment (michael-schwarz) [inline src/analyses/base.ml:1260]: Can one do better than `List.to_seq` after the append here?

Comment (michael-schwarz) [reply src/analyses/base.ml:1260]: Also, I'd prefer a space before the `@@`.

Comment (KhalilCK) [reply src/analyses/base.ml]: > Can one do better than `List.to_seq` after the append here?

Thanks for the feedback, I will try to refactor the `List.to_seq` here.

**Comment by sim642** [comment]
Compared to #974, it has this change which isn't present here: https://github.com/goblint/analyzer/pull/974/files#diff-969b2ca0b7175f19daa5deae1308a02718e8cd1927a4d8e160541a35e57c31dbL1376-R1380. Is there a reason for that?

Same with: https://github.com/goblint/analyzer/pull/974/files#diff-7c58a9f6dd84f297a28b19f30a5cd93b5a9c13c4bfe3a225c43e8a9c85b9aae5L100-R104.

**Review by michael-schwarz — APPROVED** [review]

**Review by jerhard — APPROVED** [review]

**Review by sim642 — APPROVED** [review]
The changes themselves are fine, I was just wondering about my previous point: https://github.com/goblint/analyzer/pull/975#issuecomment-1410204711.

**Comment by michael-schwarz** [comment]
We discussed this with our students offline yesterday, one of places you mentioned these was added, for the one using sort we didn't see the point as the introduction won't save anything there, as the entire list needs to be considered to sort it anyway.

**Pull request merged**